processing/src/main/java/io/druid/query/aggregation/cardinality/types/LongCardinalityAggregatorColumnSelectorStrategy.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.cardinality.types;
+
+import com.google.common.hash.Hasher;
+import io.druid.query.aggregation.cardinality.CardinalityAggregator;
+import io.druid.query.aggregation.hyperloglog.HyperLogLogCollector;
+import io.druid.segment.LongColumnSelector;
+
+public class LongCardinalityAggregatorColumnSelectorStrategy implements CardinalityAggregatorColumnSelectorStrategy<LongColumnSelector>
+{
+  @Override
+  public void hashRow(LongColumnSelector dimSelector, Hasher hasher)
+  {
+    hasher.putLong(dimSelector.get());
+  }
+
+  @Override
+  public void hashValues(LongColumnSelector dimSelector, HyperLogLogCollector collector)
+  {
+    collector.add(CardinalityAggregator.hashFn.hashLong(dimSelector.get()).asBytes());
```

Comment:
I had it convert to string because of the lack of `hashFloat`, but changed this to use int bits now, thanks